### PR TITLE
Fix included record parameter issue with `init` function

### LINKED ISSUE #27617: Issue with Included record parameter
**Description:**
if Included record parameter used in init(), it fails with Bad Sad error

**Steps to reproduce:**
```ballerina
import ballerina/io;

public function main() {
    //this works
    test(a = "test", b = 20);

    //creates a internal log and fails with Bad. Sad.
    TestClass tclass =  new(a = "sabthar", b = 23);
}

public class TestClass {
    public function init(*Test t){
        io:println(t);
    }
}

public function test(*Test t){
    io:println(t);
}

public type Test record {
    string a;
    int b;
};

```

**Affected Versions:**
slp8

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #27617 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -536,7 +536,8 @@ private void addRequiredParamsToGeneratedInitFunction(BLangFunction initFunction
                     ASTBuilderUtil.createVariable(initFunction.pos,
                                                   requiredParameter.name.getValue(), requiredParameter.type,
                                                   createRequiredParamExpr(requiredParameter.expr),
-                                                  new BVarSymbol(0, names.fromString(requiredParameter.name.getValue()),
+                                                  new BVarSymbol(Flags.asMask(requiredParameter.flagSet),
+                                                  names.fromString(requiredParameter.name.getValue()),
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:540]: Let's fix the alignment here - align this with the start of `Flags.asMask(requiredParameter.flagSet),` similar to the rest of the args below.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/functions/FunctionWithIncludedRecordParam.java [hunk 2]
```diff
@@ -185,4 +185,9 @@ public void testFunctionOfFunctionTypedParamWithIncludedRecordParam21() {
     public void testFunctionOfFunctionTypedParamWithIncludedRecordParam22() {
         BRunUtil.invoke(result, "testFunctionOfFunctionTypedParamWithIncludedRecordParam22");
     }
+
+    @Test
+    public void testFunctionOfFunctionTypedParamWithIncludedRecordParam23() {
+        BRunUtil.invoke(result, "testFunctionOfFunctionTypedParamWithIncludedRecordParam23");
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/functions/FunctionWithIncludedRecordParam.java:191]: Are these names correct? IMO, it implies there's a param of a function type.

File: tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_included_record_parameters.bal [hunk 3]
```diff
@@ -300,6 +300,21 @@ function testFunctionOfFunctionTypedParamWithIncludedRecordParam22() {
...
+    public string firstName;
+    public string secondName;
+    public function init(*NewPerson p) {
+        self.firstName = p.firstName;
+        self.secondName = p.secondName;
+    }
+}
+
+function testFunctionOfFunctionTypedParamWithIncludedRecordParam23() {
+    TestClass tclass =  new(firstName = "chiran", secondName = "sachintha");
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_included_record_parameters.bal:313]:
```suggestion
    TestClass tclass = new (firstName = "chiran", secondName = "sachintha");
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java (hunk 1 above)
Comment (chiranSachintha) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Fixed.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/functions/FunctionWithIncludedRecordParam.java (hunk 2 above)
Comment (chiranSachintha) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/functions/FunctionWithIncludedRecordParam.java]: Change as `functionWithIncludedRecordParameters`.

File: tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_included_record_parameters.bal (hunk 3 above)
Comment (chiranSachintha) [reply tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_included_record_parameters.bal]: Fixed.

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**